src/wallet/scriptpubkeyman.cpp

```diff
@@ -2280,3 +2286,38 @@ bool DescriptorScriptPubKeyMan::GetDescriptorString(std::string& out, bool priv)
 
     return m_wallet_descriptor.descriptor->ToNormalizedString(provider, out, priv);
 }
+
+void DescriptorScriptPubKeyMan::UpdateWalletDescriptor(WalletDescriptor& descriptor)
+{
+    LOCK(cs_desc_man);
+    std::string error;
+    if (!CanUpdateToWalletDescriptor(descriptor, error)) {
+        throw std::runtime_error(std::string(__func__) + ": " + error);
+    }
+
+    m_map_pubkeys.clear();
+    m_map_script_pub_keys.clear();
+    m_max_cached_index = -1;
+    m_wallet_descriptor = descriptor;
+}
+
+bool DescriptorScriptPubKeyMan::CanUpdateToWalletDescriptor(const WalletDescriptor& descriptor, std::string& error)
```

Comment:
I feel like the name of this function is confusing and could be improved.

Reply:
I've spent some time, but couldn't come up with anything better. Any suggestions?